### PR TITLE
replace validate_string with assert_type

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/addon.pp [hunk 1]
```diff
@@ -32,16 +32,17 @@
...
+  $splunk_home                   = '/opt/splunkforwarder',
+  $package_manage                = true,
+  Optional[String] $package_name = undef,
+  $inputs                        = {},
 ) {
 
 
   if ( $package_manage ) {
-    validate_string($package_name)
+    assert_type(String, $package_name)
```
Comment (juniorsysadmin) [inline manifests/addon.pp]: Shouldn't the data typing already cover this?

Comment (bastelfreak) [reply manifests/addon.pp]: ups, copy and paste failure. let me fix that.

**Comment by bastelfreak** [comment]
@juniorsysadmin fixed.

File: manifests/addon.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/addon.pp]: It *was* also allowed to be `undef`.  But if `package_manage == true` then I think it probably does have to be a string.

Comment (juniorsysadmin) [reply manifests/addon.pp]: @alexjfisher is right. But now that I think of it, what you had was better before. That accounts for the case where someone specified `$package_name` = true but did not provide a proper `$package_name`. `undef` as a value for the package resource name is also invalid in Puppet 4/5 I believe.

**Pull request merged**